impeller/renderer/backend/gles/description_gles.cc

```diff
@@ -0,0 +1,159 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/gles/description_gles.h"
+
+#include <algorithm>
+#include <cctype>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "impeller/base/strings.h"
+#include "impeller/base/validation.h"
+#include "impeller/renderer/backend/gles/proc_table_gles.h"
+
+namespace impeller {
+
+static std::string GetGLString(const ProcTableGLES& gl, GLenum name) {
+  auto str = gl.GetString(name);
+  if (str == nullptr) {
+    return "";
+  }
+  return reinterpret_cast<const char*>(str);
+}
+
+std::optional<bool> DetermineIfES(const std::string& version) {
+  if (HasPrefix(version, "OpenGL ES")) {
+    return true;
+  } else if (HasPrefix(version, "OpenGL")) {
+    return false;
+  } else {
+    return false;
+  }
+}
+
+static std::optional<Version> DetermineVersion(std::string version) {
```

Comment:
Can this have some unit tests? String parsing without unit tests makes me very nervous.

Reply:
(seconded)